streams-tests/shared/src/test/scala/zio/stream/ZStreamSpec.scala

```diff
@@ -208,6 +209,49 @@ object ZStreamSpec extends ZIOBaseSpec {
             assertM(ZStream(1, 2, 3).aggregate(t).runCollect.either)(isLeft(equalTo(fail)))
           }
         ),
+        suite("debounce")(
+          testM("drop earlier elements within waitTime") {
+            assertWithChunkCoordination(List(Chunk(1, 2), Chunk(3, 4), Chunk.single(5))) { c =>
+              val stream = ZStream
+                .fromQueue(c.queue)
+                .collectWhileSuccess
+                .flattenChunks
+                .debounce(1.second)
+                .tap(_ => c.proceed)
+
+              assertM(for {
+                f      <- stream.runCollect.fork
```

Comment:
If all testing patterns converge to "fork the stream, offer-advance-next a bunch of times then collect the results", it may be simpler to initialize the stepper with a `List[Assertion[A]]` that is tested against the incoming chunks.